python-package/lightgbm/basic.py

```diff
@@ -779,9 +779,6 @@ def predict(self, data, start_iteration=0, num_iteration=-1,
             Prediction result.
             Can be sparse or a list of sparse objects (each element represents predictions for one class) for feature contributions (when ``pred_contrib=True``).
         """
-        if isinstance(data, Dataset):
-            raise TypeError("Cannot use Dataset instance for prediction, please use raw data instead")
-        data = _data_from_pandas(data, None, None, self.pandas_categorical)[0]
```

Comment:
Why was this moved from Predictor code? Now Predictor cannot accept pandas DataFrame which means, for example, `refit()` method cannot accept DataFrames anymore:
https://github.com/microsoft/LightGBM/blob/ab9236baa06c3f071acf6b45a771c2a09661f146/python-package/lightgbm/basic.py#L3615

and DataFrames cannot be used together with `init_model` argument during constructing Dataset:
https://github.com/microsoft/LightGBM/blob/ab9236baa06c3f071acf6b45a771c2a09661f146/python-package/lightgbm/basic.py#L1402-L1404

Reply:
I agree with this comment, and will just add that it would be useful to have unit tests (in a separate PR) for them, so such regressions could be caught automatically in the future.

Reply:
Addressed in 774a715

Reply:
Thanks!

Should we add `validate_features` argument to those methods as well?